qiskit_research/utils/gate_decompositions.py

```diff
@@ -28,13 +29,42 @@
     XXPlusYYGate,
 )
 from qiskit.dagcircuit import DAGCircuit
-from qiskit.pulse.instruction_schedule_map import InstructionScheduleMap
+from qiskit.providers.backend import Backend
+from qiskit.pulse import ControlChannel, Play
 from qiskit.qasm import pi
 from qiskit.transpiler.basepasses import TransformationPass
 
 from .gates import SECRGate
 
 
+def cr_forward_direction(qc, qt, inst_sched_map, ctrl_chans) -> boolean:
```

Comment:
It looks like you typed `boolean` instead of `bool` by mistake and this caused your editor to add automatically add an erroneous import statement.

Reply:
Fixed!